dev/com.ibm.ws.org.apache.cxf.cxf.rt.transports.http.3.2/src/org/apache/cxf/transport/http/HTTPConduit.java

```diff
@@ -1684,7 +1694,9 @@ protected void handleResponseInternal() throws IOException {
             Message inMessage = new MessageImpl();
             inMessage.setExchange(exchange);
             updateResponseHeaders(inMessage);
-            inMessage.put(Message.RESPONSE_CODE, responseCode);
+            //Liberty code change start
+            ((MessageImpl) inMessage).setResponseCode(responseCode);
```

Comment:
The cast is not needed here if you update inMessage a few lines up to be of type MessageImpl instead of Mesage.